### DISCUSSION TITLE
Strange behavior of SelectiveBloomEffect when working with InstancedMesh

### DISCUSSION THREAD

**Discussion by godknowsiamgood** [pmndrs/postprocessing#520] [Q&A]
Hi, I just asked this on [here](https://discourse.threejs.org/t/strange-behavior-of-selectivebloomeffect-when-working-with-instancedmesh/54792) but thought that it is more effective to do it here as well.

So, I have a problem when regular objects are highlighted properly, but if I use InstancedMesh and change positions for instances using BufferAttribute, the effect doesn’t apply to them.
Note the strange artefact when the middle triangle is near to zero position. Like during one of the rendering passes, the shader positions setting is not applied.
Could you please give me an advice how to manage it?

Here is the simple scene where I demonstrate this: https://codesandbox.io/s/admiring-wozniak-pctkmw?file=/src/index.js

**Comment by vanruesc** [comment]
Did you enable the [override material workaround](https://github.com/pmndrs/postprocessing/wiki/Override-Materials)? This should fix depth rendering for instanced meshes which in turn fixes selective bloom.

However, please note that selecting individual instances is currently not supported.
